python/ray/train/tests/test_trainer.py

```diff
@@ -562,6 +563,25 @@ def train_func():
     assert set(results) == {0, 1}
 
 
+def test_torch_auto_unwrap(ray_start_2_cpus):
+    """Tests if underlying model from DDP is extracted when saving ckpt."""
+
+    def train_fn():
+        model = torch.nn.Linear(1, 1)
+
+        # Wrap in DDP.
+        model = train.torch.prepare_model(model)
+
+        # Save DDP wrapped model.
+        train.save_checkpoint(model=model)
+
+    num_workers = 2
+    trainer = Trainer("torch", num_workers)
+    trainer.start()
+    trainer.run(train_fn)
+    trainer.shutdown()
```

Comment:
Same here, can you access the model from `trainer.latest_checkpoint` and assert that it's not wrapped in DDP?

Reply:
If it passes all the way through, then it should be working fine otherwise it would have raised an error. But added a check just to be safe.